### PR TITLE
[DOCS] CMakeLists update for master

### PR DESCRIPTION
**Pull request by msmykx-intel**
Adjusted paths in the CMakeLists.txt file to contain new paths for code snippets.

### CODE REVIEW

File: docs/snippets/CMakeLists.txt [hunk 1]
```diff
@@ -18,8 +18,10 @@ endif()
 
 file(GLOB SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp"
                   "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp"
-                  "${CMAKE_CURRENT_SOURCE_DIR}/src/*.c")
-file(GLOB GPU_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/gpu/*.cpp")
+                  "${CMAKE_CURRENT_SOURCE_DIR}/src/*.c"
+                  "${DOCS_SOURCE_DIR}/articles_en/assets/snippets/*.cpp"
+                  "${DOCS_SOURCE_DIR}/articles_en/assets/snippets/*.c")
+file(GLOB GPU_SOURCES "${DOCS_SOURCE_DIR}/articles_en/assets/*.cpp")
```
Comment (ilya-lavrenov) [inline docs/snippets/CMakeLists.txt:24]: GPU sources does not seem to be correct.

In previous logic, sources contains generic files, while GPU sources contained only GPU specific. 
We need to preserve it.

Comment (msmykx-intel) [reply docs/snippets/CMakeLists.txt]: Moved GPU sources to separate folder, and adjusted the GPU_SOURCES line in the CMakeLists to keep the prevoius logic in https://github.com/openvinotoolkit/openvino/pull/25791/commits/4681e0536369bbf3cb7090a8ad8c9e129565a9e8

**Review by sgolebiewski-intel — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]
See https://github.com/openvinotoolkit/openvino/pull/25802#discussion_r1697075273

We need to ensure that all snippets files are build during that build stage.

**Comment by ilya-lavrenov** [comment]
@olpipi 
Could you please help to fix code snippets? 
For some reason, checks for code snippets were disabled temporary and some of docs code snippets were broken.

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**